OpenSim/Simulation/Model/ModelComponent.cpp

```diff
@@ -76,7 +79,16 @@ void ModelComponent::extendConnect(Component &root)
         connectToModel(*model);
 }
 
-
+void ModelComponent::extendFinalizeFromProperties()
+{
+    Super::extendFinalizeFromProperties();
+    int geomSize = getProperty_GeometrySet().size();
+    if (geomSize > 0){
+        for (int i = 0; i < geomSize; ++i){
+            addComponent(&upd_GeometrySet(i));
```

Comment:
@aseth1 is this an OK use of `addComponent()`? It is in `finalizeFromProperties()` so gets called from the constructor.